lib/media/drm_engine.js

```diff
@@ -940,6 +940,9 @@ shaka.media.DrmEngine = class {
           }
           this.destroyer_.ensureNotDestroyed();
         } catch (e) {
+          if (e instanceof shaka.util.Error) {
```

Comment:
Hm.
I spent a little bit looking at this, wondering if it would be better to just move the code that checks `status` so it happens after the try-catch block. It feels a little weird to produce an error inside the try-catch block, then handle it by just immediately throwing it again.
In the end I think this approach is probably the best, though. I could imagine someone adding some extra function call into the try block in the future, which can also throw a Shaka error, and not noticing the potential problem.